stdlib/public/TensorFlow/Dataset.swift

```diff
@@ -182,23 +182,25 @@ extension DatasetIterator : IteratorProtocol {
   public mutating func next() -> Element? {
     let optional: VariantHandle =
       #tfop("IteratorGetNextAsOptional", _handle,
-            output_types$typeList: Element.self,
-            output_shapes$unknownShapeList: Element.self)
+            output_types$dtype: Element._outputTypeList,
+            output_shapes: Element._unknownShapeList)
     guard _TFGetScalarOrDie(#tfop("OptionalHasValue", optional)) else {
       return nil
     }
     return #tfop("OptionalGetValue", optional,
-                 output_types$typeList: Element.self,
-                 output_shapes$unknownShapeList: Element.self) as Element
+                 output_types$dtype: Element._outputTypeList,
+                 output_shapes: Element._unknownShapeList) as Element
   }
 }
 
 // FIXME: SR-8599 blocks heterogeneous scalar types.
-@inlinable @inline(__always)
-public func zip<T>(
-  _ dataset1: Dataset<T>, _ dataset2: Dataset<T>
-) -> Dataset<(T, T)> {
-  return #tfop("ZipDataset", [dataset1, dataset2],
-               output_types$typeList: (T, T).self,
-               output_shapes$unknownShapeList: (T, T).self)
-}
+// FIXME: Tuples can't conform to TensorGroup, so we will have to define a
```

Comment:
You can define a `TensorPair` struct for this. Tuples datasets are extremely common so we should support it.

Reply:
done

Reply:
oops, pasted "done" into the wrong box :)

Will actually do as a separate PR, as mentioned in my top-level comment.